src/decomp_2d.f90

```diff
@@ -208,12 +208,31 @@ module decomp_2d
   integer, save :: iskipV, jskipV, kskipV
   integer, save :: iskipP, jskipP, kskipP
 
+  !
+  ! Profiler section
+  !
+  ! Integer to select the profiling tool
+  !    0 => no profiling, default
+  !    1 => Caliper (https://github.com/LLNL/Caliper)
+  !
+  enum, bind(c)
+    enumerator :: decomp_profiler_none = 0
+    enumerator :: decomp_profiler_caliper = 1
+  end enum
+  integer(kind(decomp_profiler_none)), save, public :: decomp_profiler = decomp_profiler_none
+  ! Default : profile everything
+  logical, save, public :: decomp_profiler_transpose = .true.
+  logical, save, public :: decomp_profiler_io = .true.
+  logical, save, public :: decomp_profiler_fft = .true.
+  logical, save, public :: decomp_profiler_d2d = .true.
+
   ! public user routines
   public :: decomp_2d_init, decomp_2d_finalize, &
        transpose_x_to_y, transpose_y_to_z, &
        transpose_z_to_y, transpose_y_to_x, &
        decomp_info_init, decomp_info_finalize, partition, &
-       decomp_info_print, &
+       decomp_info_print, decomp_profiler_prep, &
+       decomp_profiler_start, decomp_profiler_end, &
```

Comment:
Do these need to be #ifdef'd out? Are they defined if `PROFILER=none`?

Reply:
The interface is defined yes